src/Platform/Vulkan/VulkanUniformManager.cc

```diff
@@ -0,0 +1,285 @@
+        VK_SUCCESS)
+    {
+      ESP_CORE_ERROR("Failed to allocate descriptor sets!");
+      throw std::runtime_error("Failed to allocate descriptor sets!");
+    }
+  }
+
+  void VulkanDescriptorSetPacgake::bind_data_and_descriptors(std::vector<VulkanMetaUniform>& uniforms,
+                                                             VkDescriptorSet& ds,
+                                                             DescriptorSetBuffers& ds_data)
+  {
+    std::vector<VkWriteDescriptorSet> descriptorWrites;
+    std::vector<VkDescriptorBufferInfo> bufferInfos;
+
+    for (auto& uniform : uniforms)
+    {
+      if (uniform.m_uniform_type == EspUniformType::ESP_BUFFER_UNIFORM)
+      {
+        VkDescriptorBufferInfo bufferInfo{};
+        bufferInfo.buffer = ds_data.m_buffers[uniform.m_binding]->get_buffer();
+        bufferInfo.offset = 0;
+        bufferInfo.range  = uniform.m_size_of_data_chunk * uniform.m_count_of_elements;
+        bufferInfos.push_back(bufferInfo);
+
+        VkWriteDescriptorSet descriptorWrite{};
+        descriptorWrite.sType           = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+        descriptorWrite.dstSet          = ds;
+        descriptorWrite.dstBinding      = uniform.m_binding;
+        descriptorWrite.dstArrayElement = 0;
+        descriptorWrite.descriptorType  = VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER;
+        descriptorWrite.descriptorCount = uniform.m_count_of_elements;
+        descriptorWrite.pBufferInfo     = &bufferInfos.back();
+
+        descriptorWrites.push_back(descriptorWrite);
+      }
+      else if (uniform.m_uniform_type == EspUniformType::ESP_TEXTURE)
+      {
+        ESP_CORE_ERROR("NO IMPLEMENTED TEXTURES!");
+        throw std::runtime_error("NO IMPLEMENTED TEXTURES!");
+      }
+    }
+
+    vkUpdateDescriptorSets(VulkanDevice::get_logical_device(),
+                           static_cast<uint32_t>(descriptorWrites.size()),
+                           descriptorWrites.data(),
+                           0,
+                           nullptr);
+  }
+
+  void VulkanDescriptorSetPacgake::write(int set, int binding, int size, void* data)
+  {
+    m_buffer_groups[set].m_buffers[binding]->write_to_buffer(data, size);
+  }
+
+  void VulkanDescriptorSetPacgake::write(int set, int binding, int elem, int size, void* data)
+  {
+    m_buffer_groups[set].m_buffers[binding]->write_to_buffer(data, size, elem * size);
+  }
+
+} // namespace esp
+
+namespace esp
+{
+  VulkanUniformManager::VulkanUniformManager(std::unique_ptr<VulkanUniformMetaData> meta_data) :
+      m_meta_data(std::move(meta_data))
+  {
+    /*
+      SHORTCUTS:
+        dsl - Descriptor Set Layout
+        dslbs - Descriptor Set Layout Bindings
+    */
+    m_ds_layouts = std::make_unique<std::vector<VkDescriptorSetLayout>>();
+
+    int counter_dsl = 1;
+    for (auto& meta_dsl : m_meta_data->m_meta_descriptor_sets)
+    {
+      std::vector<VkDescriptorSetLayoutBinding> dslbs;
```

Comment:
awugydasawe:D

Reply:
However, I would like to point out that the lack of physical contact between us does not imply a lack of respect for personal etiquette @sebastianSzarafin